core/src/main/java/hudson/lifecycle/WindowsServiceLifecycle.java

```diff
@@ -144,6 +145,15 @@ public void restart() throws IOException, InterruptedException {
         if(r!=0)
             throw new IOException(baos.toString());
     }
+    
+    private static final File getBaseDir() throws FileNotFoundException {
+        String baseEnv = System.getenv("BASE");
+        if (baseEnv == null) {
+            throw new FileNotFoundException("Unable to locate Jenkins base directory. Environment variable 'BASE' not found.");
```

Comment:
I would still try Jenkins root dir as a fallback. Otherwise there is a risk of breaking some cases.